frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -2042,8 +2156,13 @@ function ReaderHighlight:getSavedExtendedHighlightPage(hl_or_bm, page, index)
 end
 
 function ReaderHighlight:onReadSettings(config)
+    G_reader_settings:initializeExtSettings("highlight_colors", default_highlight_colors)
```

Comment:
I think we'd rather NOT have the current defaults saved into settings.reader.lua and get them be used forever instead of any future adjustment we may do to those defaults - which I think is what initializeExtSettings do.
Only later when/if there is a dialog to allow tuning them, this dialog should save them into settings.reader.lua.

Reply:
My motivation was to allow someone to use a texteditor to edit the builtin values.
I hadn't considered the update implications (not being able to push future adjustments).

Maybe the best way is to have the option to read the key (if set) but with no default.
Then we can make a wiki page documenting the syntax if someone really wants different colors without a dialog.

Reply:
Done.
https://github.com/koreader/koreader/compare/419a04d84b0a14710a98e7bf5e05ec4ac2077e3b..bb746b20dfe9ccf9108c3d83ed85c8683fbaeecc

Reply:
If you really want to allow easier user tweaking of these colors, I think it's best if it is done with a [user patch](https://github.com/koreader/koreader/wiki/User-patches), than with editing settings.reader.lua.
The fact that you set these defaults as a local module variable `local default_highlight_colors` will make it [harder to override them from a user patch](https://github.com/koreader/koreader/pull/9104#issuecomment-1509160619).
If they were set in `InputContainer:extend{}`, one could just override them with
```lua
local ReaderHighlight = require("apps/reader/modules/readerhighlight")
ReaderHighlight.default_highlight_colors = {
    {_("Red"), "#FF0000"},
    {_("Orange"), "#ffAA00"},
}
```
But currently, there's nothing set in ReaderHighlight's `InputContainer:extend{}`, so it feels a bit odd to go add that in there :/
(The other `local` stuff are mostly translatable strings or keys, that don't need to be available for tweaking.)

Reply:
Thanks for the pointer. I had assumed patches would require recompiling from source (ala "suckless" software).
Live patches worked fine on android after making /sdcard/koreader/patches/2-highlight-colors.lua.
I got an "Error loading patch" because `_` was undefined. But adding an import for that fixed everything.

```
local _ = require("gettext")
local ReaderHighlight = require("apps/reader/modules/readerhighlight")
ReaderHighlight.highlight_colors = {
    {_("Red"), "#fe4400"},
    {_("Orange"), "#ff8800"}
}
```

I don't think adding `highlight_colors` to `InputContainer:extend{}` is too crazy of an idea. The resulting diff is quite small.

Reply:
> I don't think adding `highlight_colors` to `InputContainer:extend{}` is too crazy of an idea. The resulting diff is quite small.

Ouch, please no.

We have a *lot* of InputConbtainers, and very few of them will actually care about that ;).

Reply:
Err, nuless you were actually talking about *another* object extending InputContainer? ;D.

Reply:
Yes, it would be another object: ReaderHighlight, if we do:
```diff
-local ReaderHighlight = InputContainer:extend{}
+local ReaderHighlight = InputContainer:extend{
+    -- Make that available to user-patches
+    default_highlight_colors = {
+        {_("Red"), "#FF0000"},
+        {_("Orange"), "#ffAA00"},
+    }
}
```